test/distributed/_tensor/test_utils.py

```diff
@@ -127,6 +127,30 @@ def test_compute_local_shape_and_global_offset_2D(self):
                     global_tensor[dim0_start:dim0_end, dim1_start:dim1_end],
                 )
 
+    @with_comms
+    def test_fsdp_tp_meta_compute(self):
+        # FSDP + TP sharding
+        tp_size = 2
+        dp_size = self.world_size // tp_size
+        global_mesh = init_device_mesh(
+            self.device_type, (dp_size, tp_size), mesh_dim_names=("dp", "tp")
+        )
+        # local shard shape is [2, 2]
+        global_tensor_shape = torch.Size([2 * self.world_size, 2])
+        placements = [_StridedShard(0, split_factor=tp_size), Shard(0)]
```

Comment:
Can you also add a test case of [Replicate(), _StridedShard, Shard], this would cover HSDP state dict offset calculation